.github/workflows/build-and-test.yml

```diff
@@ -32,3 +32,7 @@ jobs:
       - name: Run tests
         run: pnpm test
         timeout-minutes: 5
+
+      - name: Validate engineering documentation
+        run: pnpm test:docs
+        timeout-minutes: 2
```

Comment:
Adding a new step to CI, works in this PR - you can view it in the `build-and-test` job.